extensions/quartz/deployment/src/main/java/io/quarkus/quartz/deployment/QuartzProcessor.java

```diff
@@ -118,19 +126,50 @@ QuartzJDBCDriverDialectBuildItem driver(List<JdbcDataSourceBuildItem> jdbcDataSo
                     "The Agroal extension is missing and it is required when a Quartz JDBC store is used.");
         }
 
-        Optional<JdbcDataSourceBuildItem> selectedJdbcDataSourceBuildItem = jdbcDataSourceBuildItems.stream()
-                .filter(i -> config.dataSourceName.isPresent() ? config.dataSourceName.get().equals(i.getName())
-                        : i.isDefault())
-                .findFirst();
-
-        if (!selectedJdbcDataSourceBuildItem.isPresent()) {
-            String message = String.format(
-                    "JDBC Store configured but the '%s' datasource is not configured properly. You can configure your datasource by following the guide available at: https://quarkus.io/guides/datasource",
-                    config.dataSourceName.isPresent() ? config.dataSourceName.get() : "default");
-            throw new ConfigurationException(message);
+        Optional<String> driverDelegate = config.driverDelegate;
+        if (driverDelegate.isPresent()) {
+            // user-specified custom delegate
+            IndexView indexView = indexBuildItem.getIndex();
+            ClassInfo customDelegate = indexView.getClassByName(driverDelegate.get());
+            if (customDelegate == null) {
+                String message = String.format(
+                        "Custom JDBC delegate implementation class '%s' was not found in Jandex index. " +
+                                "Make sure the dependency containing this class has proper marker file enabling discovery. " +
+                                "Alternatively, you can index a dependency using IndexDependencyBuildItem.",
+                        driverDelegate.get());
+                throw new ConfigurationException(message);
+            } else {
+                // any custom implementation needs to be a subclass of known Quarkus delegate
+                boolean implementsKnownDelegate = false;
+                for (DotName knownImplementation : Set.of(DELEGATE_MSSQL, DELEGATE_POSTGRESQL, DELEGATE_DB2V8, DELEGATE_STDJDBC,
+                        DELEGATE_HSQLDB)) {
+                    if (indexView.getAllKnownSubclasses(knownImplementation).contains(customDelegate)) {
```

Comment:
You should not use `Collection#contains()` for `ClassInfo` because it does not redefine `equals()`, i.e. object identity is used.

Reply:
Ah, right. `DotName` it is then.